proposal.md

```diff
@@ -0,0 +1,137 @@
+## Flexible PipelineAgent IO
+
+Rename ``VoicePipelineAgent`` to ``PipelineAgent``
+
+This API is non-breaking (backward compatible)
+
+STT -> LLM -> TTS
+
+```python
+AudioInput = AsyncIterator[rtc.AudioFrame | rtc.AudioFrameEvent]
+
+class AudioOutput(Protocol):
+    async def capture_frame(self, frame: rtc.AudioFrame) -> None: ...
+
+    def flush(self) -> None: ...
+
+    def clear_queue(self) -> None: ...
+
+
+class TextOutput(Protocol):
+    async def write(self, text: str) -> None: ...
+
+    def flush(self) -> None: ...
+
+
+class PipelineIO(ABC):
```

Comment:
this name feels a little odd, given that in addition to `PipelineIO` we also have `PipelineOutput` (and maybe some forthcoming `Input` protocol too), but the "IO" in `PipelineIO` is not related to the pipeline's input nor output itself... might be better as `PipelineHooks`

Reply:
I like `PipelineHooks` as a name